### PR TITLE
Adding P2SH(p2pkh) script test case

### PR DESCRIPTION
**Pull request by Christewart**
The happy path for a p2sh script with redeem script type p2pkh was previously untested, adding a test case for this inside of script_tests.json

### CODE REVIEW

**Comment by paveljanik** [comment]
Please squash into one commit (see https://github.com/bitcoin/bitcoin/pull/8034/commits/90963e53707cf244e6a2a804341e9280628b0c34#diff-6a3371457528722a734f3c51d9238c13R46).

**Comment by Christewart** [comment]
Not sure why Travis CI failed on this build - I don't know Travis CI that well but it looks like it is unrelated to my change?

**Comment by sdaftuar** [comment]
Agree that the failure looks unrelated.  But yikes, that looks like a somewhat scary error that was stumbled upon, will investigate...

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/8090/commits/32833a4d96023bdbc28ae1e9d441cfdb1e2f9e8e

Travis failure is indeed unrelated.

**Comment by Christewart** [comment]
@paveljanik @sdaftuar Is there a way to restart this travis build?

**Comment by arowser** [comment]
Can one of the admins verify this patch?

**Comment by paveljanik** [comment]
Lokks like it is already done - all checks have passed.

**Comment by sipa** [comment]
@Christewart Please don't add manually created test cases inside the "automically generated test cases" block, as that block gets overwritten.

Ideally, add this as an automatically generated test here: https://github.com/bitcoin/bitcoin/blob/v0.12.1/src/test/script_tests.cpp#L353

File: src/test/script_tests.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@
 using namespace std;
 
 // Uncomment if you want to output updated JSON tests.
-// #define UPDATE_JSON_TESTS
+//#define UPDATE_JSON_TESTS
```
Comment (laanwj) [inline src/test/script_tests.cpp]: This should not be changed

**Comment by laanwj** [comment]
Sorry for this taking so long, just moving the test case would have been fine, but now that you change script_tests.cpp you need to make sure it doesn't generate the new json every time.

**Comment by Christewart** [comment]
@laanwj Should be fixed - I always seem to add unnecessary spaces/new lines without realizing it in my commits :-)

**Comment by sipa** [comment]
utACK b682960a28137966306e0a3646c96fa5f4b811ff

File: src/test/script_tests.cpp [hunk 2]
```diff
@@ -427,7 +427,10 @@ BOOST_AUTO_TEST_CASE(script_build)
     tests.push_back(TestBuilder(CScript() << ToByteVector(keys.pubkey0C) << OP_CHECKSIG,
                                 "P2SH(P2PK), bad redeemscript", SCRIPT_VERIFY_P2SH, true
                                ).PushSig(keys.key0).PushRedeem().DamagePush(10).ScriptError(SCRIPT_ERR_EVAL_FALSE));
-
+
```
Comment (paveljanik) [inline src/test/script_tests.cpp]: Pure cosmetic nit: you have added tab here.

**Comment by paveljanik** [comment]
utACK https://github.com/bitcoin/bitcoin/commit/b682960a28137966306e0a3646c96fa5f4b811ff

File: src/test/script_tests.cpp (hunk 2 above)
Comment (laanwj) [inline src/test/script_tests.cpp]: Nit^2: I think it's not a tab but four spaces :snail:

**Comment by laanwj** [comment]
Not going to hold this up on a cosmetic nit.

**Pull request merged**